core/src/main/java/org/axonframework/serialization/compression/GzipSerializerWrapper.java

```diff
@@ -0,0 +1,48 @@
+package org.axonframework.serialization.compression;
+
+import org.axonframework.serialization.Serializer;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+import java.util.zip.ZipException;
+
+/**
+ * AbstractSerializerCompressionWrapper implementation that used gzip compression
+ *
+ * @author Michael Willemse
+ */
+public class GzipSerializerWrapper extends AbstractSerializerCompressionWrapper {
+
+    public GzipSerializerWrapper(Serializer embeddedSerializer) {
+        super(embeddedSerializer);
+    }
+
+    @Override
+    protected byte[] doCompress(byte[] uncompressedData) throws IOException {
+        try (ByteArrayOutputStream bos = new ByteArrayOutputStream(uncompressedData.length);
+             GZIPOutputStream gzipOS = new GZIPOutputStream(bos)) {
+            gzipOS.write(uncompressedData);
+            gzipOS.close();
+            return bos.toByteArray();
+        }
+    }
+
+    @Override
+    protected byte[] doDecompress(byte[] compressedData) throws IOException {
+        try (ByteArrayInputStream bis = new ByteArrayInputStream(compressedData);
```

Comment:
With GZip, it's actually easy to validate whether a byte[] contains zipped information: the first two bytes should be `0x1f8b`. If they aren't, data is probably already decompressed (or never compressed). The wrapper could then just return the original byte[].